roles/dnsmasq/tasks/manage_host.yml

```diff
@@ -0,0 +1,58 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Assert we have needed host data
+  ansible.builtin.assert:
+    quiet: true
+    that:
+      - cifmw_dnsmasq_host_network is defined
+      - cifmw_dnsmasq_host_state is defined
+      - cifmw_dnsmasq_host_state in ['present', 'absent']
+      - cifmw_dnsmasq_host_mac is defined
+      - cifmw_dnsmasq_host_ipv6 is defined or
+        cifmw_dnsmasq_host_ipv4 is defined
+
+- name: Check network file status
+  register: _network_exists
+  ansible.builtin.stat:
+    path: "{{ cifmw_dnsmasq_basedir }}/{{ cifmw_dnsmasq_host_network }}.conf"
+    get_attributes: false
+    get_checksum: false
+    get_mime: false
+
+- name: Assert network exists
+  ansible.builtin.assert:
+    that:
+      - _network_exists.stat.exists
+    quiet: true
+    msg: >-
+      You must create the network first, calling
+      dnsmasq/manage_network.yml task.
+
+- name: Manage host entry
+  become: true
+  notify: Restart dnsmasq
+  ansible.builtin.blockinfile:
+    block: |
+      {% if cifmw_dnsmasq_host_ipv4 is defined -%}
+      dhcp-host={{ cifmw_dnsmasq_host_mac }},{{ cifmw_dnsmasq_host_ipv4 }},1h
```

Comment:
Can do in a follow up, but it would be neat to add a label here as well. For example: `set:{{ libvirt_vm_uuid }}` or if we don't have the uuid yet and vm name is uniq we can just set the name.
By doing that we open the door to add per-host dhcp options.

Reply:
we have a unique VM name. But here again, we might need to create two tags? _v4 and _v6?